### DISCUSSION TITLE
Are there logs for what Tribler participates in?

### DISCUSSION THREAD

**Discussion by balupton** [Tribler/tribler#5777] [Q&A]
Tribler differs from normal torrent clients, which only download/seed/transfer the torrents you've selected. Tribler will be part of a p2p network and share data with other Tribler users, or if Tribler is configured to be an exit node, then Tribler will download torrents on other Tribler users behalf, and seed it to them, thus facilitating anonymity.

Is there any logs anywhere so we can know what content Tribler is exchanging on behalf of others?

**Comment by qstokkink** [comment]
Hi, before I give you the answer let me make sure it is clear who can see what. There are two types of anonymization in Tribler:

 1. Hidden services (complete e2e encryption).
 2. Exit nodes (connection to the wild Internet Bittorrent peers).

Besides the exit nodes, the seeders and the leechers, none of the intermediate peers in the anonymization can see anything.

![tunnelvisibility](https://user-images.githubusercontent.com/3630389/100980533-a60fc900-3545-11eb-9f76-78d0157091c6.png)

 If you _are_ running an exit node (if so, be careful - this may be illegal depending on where you live) we log the IPs you connect to on others' behalf. If you set the Tribler logger to `DEBUG` level, your log will contain statements like `Resolved hostname http://sometracker.com/ to ip_address (1.2.3.4, 5)`. We do not log what is being downloaded though, you would need to modify the code yourself.
